src/components/primitive/button/button.component.tsx

```diff
@@ -1,10 +1,47 @@
-import { ButtonProps } from './button.component.props';
-import { StyledButton } from './button.component.styles';
+import styled from 'styled-components';
+import { IconSvg } from '../../primitive';
+import { Primary, Small, GhostButton, IconWrapper, ButtonText } from './button.component.styles';
+import { ButtonComponentProps, Variant } from './button.component.props';
+
+export const Button: React.FunctionComponent<ButtonComponentProps> = (props: ButtonComponentProps) => {
+  const { variant, label, icon, onClick, ...rest } = props;
+
+  const getVariant = (variant: Variant = Variant.primary) => {
+    let ButtonComponent;
+    switch (variant) {
+      case Variant.primary:
+        ButtonComponent = Primary;
+        break;
+      case Variant.small:
+        ButtonComponent = Small;
+        break;
+      case Variant.ghost:
+        ButtonComponent = GhostButton;
```

Comment:
Button naming is inconsistent. Append the word `Button` for all the buttons. Also proper line-breaks for this function.